test/integration/http_integration.cc

```diff
@@ -247,20 +247,19 @@ void HttpIntegrationTest::cleanupUpstreamAndDownstream() {
   }
 }
 
-uint64_t
-HttpIntegrationTest::waitForNextUpstreamRequest(uint64_t upstream_index,
-                                                absl::optional<uint64_t> second_upstream_index) {
-  uint64_t upstream_with_request = upstream_index;
+uint64_t HttpIntegrationTest::waitForNextUpstreamRequest(
+    const std::unordered_set<uint64_t>& upstream_indices) {
```

Comment:
I think you want this to be a vector or you're going to have a source of non-determinism leading to test flakes.

Reply:
Ah that's a good point, I'll change